src/main/java/com/iexec/core/task/update/TaskUpdateRequestManager.java

```diff
@@ -83,8 +85,7 @@ public void consumeAndNotify() {
             log.info("Waiting requests from publisher [queueSize:{}]", queue.size());
             try {
                 String chainTaskId = queue.take();
-                CompletableFuture.runAsync(() ->
-                        consumer.onTaskUpdateRequest(chainTaskId));
+                taskUpdateLocks.runAsyncWithLock(chainTaskId, () -> consumer.onTaskUpdateRequest(chainTaskId));
```

Comment:
I think these comment https://github.com/iExecBlockchainComputing/iexec-core/blob/32a65c1d096c8fc62d5f58b2f378240b653949e4/src/main/java/com/iexec/core/task/TaskUpdateManager.java#L78 and method name are quite wrong https://github.com/iExecBlockchainComputing/iexec-core/blob/32a65c1d096c8fc62d5f58b2f378240b653949e4/src/main/java/com/iexec/core/task/TaskUpdateManager.java#L91

Reply:
I disagree. As far as I understand the comment, it refers to this line:
https://github.com/iExecBlockchainComputing/iexec-core/blob/4ab44596d925096cf5e744cb9eef350a274e6210/src/main/java/com/iexec/core/task/update/TaskUpdateRequestManager.java#L86
It is called asynchronously, but it is not an asynchronous function. So we are sure it'll be finished in the new version once `TargetedLock#runWithLock` returns:
https://github.com/iExecBlockchainComputing/iexec-core/blob/32a65c1d096c8fc62d5f58b2f378240b653949e4/src/main/java/com/iexec/core/utils/TargetedLock.java#L36

It may not have to be placed there, but it's not factually false.

Reply:
I agree, comment should be elsewhere & updateTaskRunnable(..) is not a runnable

Reply:
Right. I have removed the comment and renamed the method. It is now called `updateTask`.
I think it was named `updateTaskRunnable` because `updateTask` was already taken by the method now called `publishUpdateTaskRequest`. It looks better now :)

Reply:
I think it was called updateTaskRunnable because it used to run in a java TaskExecutor.